### PR TITLE
ADDEED API  For employee With out test

### PR DESCRIPTION
**Pull request by sobinsunny**

### CODE REVIEW

**Comment by tachyons** [comment]
Use rubocop -a to fix formatting errors

**Comment by tachyons** [comment]
And please rebase the repository

File: app/controllers/api/v1/employees_controller.rb [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+
+      def  destroy
+      	responce = Employee.update_attributes(:is_active=>false)
+      	render json: {:responce=>responce}
+      end
+      
+  	private
+
+     def render_response
+  	 	render json: @results
```
Comment (tachyons) [inline app/controllers/api/v1/employees_controller.rb]: Do we really need to specify json here ? We have json as default in routes

File: app/controllers/api/v1/employees_controller.rb [hunk 2]
```diff
@@ -1,10 +1,39 @@
 module Api
   module V1
     class EmployeesController < ApiController
-      before_action :doorkeeper_authorize!
+      # before_action :doorkeeper_authorize!
+       # after_filter :render_response, unless: -> { @results.nil? }
       def index
-        render json: Employee.active
+        @results = Employee.active
```
Comment (tachyons) [inline app/controllers/api/v1/employees_controller.rb:7]: I think it is better to remove instance variable here

File: app/controllers/api/v1/employees_controller.rb [hunk 3]
```diff
@@ -1,10 +1,39 @@
 module Api
   module V1
     class EmployeesController < ApiController
-      before_action :doorkeeper_authorize!
+      # before_action :doorkeeper_authorize!
```
Comment (tachyons) [inline app/controllers/api/v1/employees_controller.rb:4]: Do not comment authorization filter !!

File: app/serializers/employee_serializer.rb [hunk 4]
```diff
@@ -27,5 +27,5 @@
 #
 
 class EmployeeSerializer < ActiveModel::Serializer
-  attributes :id, :employee_id,:name,:email,:date_of_birth
+  attributes :id, :employee_id, :name, :email, :date_of_birth
```
Comment (tachyons) [inline app/serializers/employee_serializer.rb:30]: Add department and other fields

File: Gemfile [hunk 5]
```diff
@@ -90,9 +90,9 @@ gem 'rails-observers'
 gem 'acts_as_pluggable' # , github: 'tachyons/acts_as_pluggable'
 # gem 'acts_as_pluggable',path: '~/projects/acts_as_pluggable'
 
-Dir.entries('plugins').select { |f| !File.directory? f }.each do |plugin|
-  gem plugin, path: 'plugins/' + plugin
-end
+# Dir.entries('plugins').select { |f| !File.directory? f }.each do |plugin|
```
Comment (tachyons) [inline Gemfile]: Why ?

Comment (tachyons) [inline Gemfile]: Please revert this!

File: app/controllers/api/v1/employees_controller.rb (hunk 3 above)
Comment (tachyons) [inline app/controllers/api/v1/employees_controller.rb:4]: Please revert this

File: app/serializers/employee_serializer.rb [hunk 6]
```diff
@@ -27,5 +27,5 @@
 #
 
 class EmployeeSerializer < ActiveModel::Serializer
-  attributes :id, :employee_id, :department_id
+  attributes :id, :employee_id, :name, :email, :date_of_birth
```
Comment (tachyons) [inline app/serializers/employee_serializer.rb]: add department also

**Comment by tachyons** [comment]
Build failing , seems like you forgot to don bundle install/update after changing gemfile

**Comment by sobinsunny** [comment]
@tachyons  please  commit that gem  file issue. I am in bus now. Limited  network support

File: app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -29,6 +29,7 @@
 class EmployeesController < ApplicationController
    before_action :authenticate_user!
    before_action :set_employee, only: [:show, :edit, :update, :destroy, :change_manager]
+   before_acti`on :authenticate_user!
```
Comment (sobinsunny) [inline app/controllers/employees_controller.rb]: Fix commited

**Comment by sobinsunny** [comment]
implimented - basic api structure

**Pull request merged**